src/libgap-api.c

```diff
@@ -0,0 +1,127 @@
+// LibGAP API - API for using GAP as shared library.
+
+#include "libgap-api.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "gapstate.h"
+#include "gvars.h"
+#include "intobj.h"
+#include "opers.h"
+#include "plist.h"
+#include "stringobj.h"
+
+Obj FuncREAD_ALL_COMMANDS(Obj self, Obj instream, Obj echo, Obj capture, Obj resultCallback);
+
+// Setup and initialisation
+
+void GAP_Initialize(int          argc,
+                    char **      argv,
+                    char **      env,
+                    CallbackFunc markBagsCallback,
+                    CallbackFunc errorCallback)
+{
+    InitializeGap(&argc, argv, env);
+    ExtraMarkFuncBags = markBagsCallback;
+    STATE(JumpToCatchCallback) = errorCallback;
+}
+
+void GAP_Finalize(void)
+{
+    FinishBags();
```

Comment:
The intention is to finish the GAP "session", and be able to `GAP_Initialize` again, having a fresh GAP session. 

Of course better yet, in future, if we have all state encapsulated, we'd be able to run multiple GAPs in one process.

That said, if noone sees a use for this, we can just leave it out for now.

Reply:
I do not think we would need this in OSCAR for now, so I think it could be removed.

Reply:
I'd quite like to be able to finalize GAP for my racket bindings, otherwise using libgap in drracket is a bit of a pain, as it crashes on reload. I might be able to address this issue by unloading the .so file, but have to check whether that's possible.

Either way, its for me to implement if noone else wants it.